app/views/layouts/application.html.erb

```diff
@@ -2,13 +2,88 @@
 <html>
 <head>
   <title>Betsy</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
+  <meta charset="utf-8">
+
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
+
+  <%=stylesheet_link_tag "application"%>
+
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>
+  <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
 </head>
 <body>
+  <main class="container">
+    <nav class="navbar navbar-default">
+      <div class="container-fluid">
+        <div class="navbar-header"><%=link_to "Betsy", root_path, class:"navbar-brand"%></div>
+        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+          <div class="col-lg-6">
+            <%= form_tag(search_product_path, method: "get", class: "navbar-form navbar-left") do %>
+            <div class="input-group">
+              <%= text_field_tag(:q, nil, class: 'form-control', placeholder: "Search a product") %>
+              <span class="input-group-btn">
+                <%= submit_tag("Search", class: "btn btn-default") %>
+                <% end %>
+              </span>
+            </div>
+          </div>
+          <ul class="nav navbar-nav navbar-right">
+            <li class="active"><%= link_to "All Products", products_path%></li>
+
+            <% if !current_user.nil? && current_user.id != nil %>
```

Comment:
Can you have a `current_user` with a `nil` value for `id`?

Extension idea:
Try this
`<% if current_user %>`

If it breaks things see if you can change how `current_user` works so that it doesn't break things.